packages/paste-core/components/badge/src/hooks.tsx

```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+import type {Padding} from '@twilio-paste/style-props';
+import {Button} from '@twilio-paste/button';
+import {Box} from '@twilio-paste/box';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+
+import type {BadgeProps} from './types';
+import {hasValidAnchorVariantProps, hasValidButtonVariantProps} from './utils';
+
+const DEFAULT_ICON_SIZE = 'sizeIcon10';
+const BADGE_ICON_PADDING = 'space10';
+const FOCUSABLE_STYLES = {
+  textDecoration: 'underline',
+  cursor: 'pointer',
+  _hover: {textDecoration: 'none'},
+  _focus: {boxShadow: 'shadowFocus', textDecoration: 'none'},
+};
+
+export type Wrapper = React.FC<{children: React.ReactNode}>;
+export type StyleProps =
+  | {
+      textDecoration: 'underline';
+      cursor: 'pointer';
+      _hover: {textDecoration: 'none'};
+      _focus: {boxShadow: 'shadowFocus'; textDecoration: 'none'};
+    }
+  | Record<string, never>;
+
+export const useFocusableVariants = ({
+  as,
+  href,
+  onClick,
+}: Pick<BadgeProps, 'as' | 'onClick' | 'href'>): {
+  wrapper: Wrapper;
+  styleProps: StyleProps;
+} => {
+  const isValidButtonVariant = hasValidButtonVariantProps({as, onClick});
+  const isValidAnchorVariant = hasValidAnchorVariantProps({as, href});
+
+  const styleProps = isValidAnchorVariant || isValidButtonVariant ? {...FOCUSABLE_STYLES} : {};
+
+  // eslint-disable-next-line unicorn/consistent-function-scoping
+  let wrapper: Wrapper = ({children}) => <>{children}</>;
+
+  if (isValidButtonVariant) {
+    wrapper = ({children}) => (
+      <Button variant="reset" size="reset" onClick={onClick}>
+        {children}
+      </Button>
+    );
+  } else if (isValidAnchorVariant) {
+    wrapper = ({children}) => (
+      <Button variant="reset" size="reset" as="a" href={href}>
+        {children}
+      </Button>
+    );
+  }
+
+  return {
+    styleProps,
+    wrapper,
+  };
+};
+
+const StyledSpan: React.FC<{children: React.ReactNode}> = ({children, ...props}) => (
+  <Box display="inherit" as="span" alignSelf="center" {...props}>
+    {children}
+  </Box>
+);
+
+export const useWrapChildren = (reactChildren: React.ReactNode): React.ReactNode => {
```

Comment:
Are we taking inspiration from the button component in here? I'm wondering if there is a CSS solution to simplify things. I might have a play tomorrow

Reply:
Do you mean mapping through the children (if there are children and not a single child) and wrapping them, etc.? 
Not explicitly no. I opted to use `Box` as a `span` and style the span through the props to be more javascripty (since we are using css in js) and composability using the Paste primitives.

Other than being verbose, what do you think is(are) the cost(s) of this compared to pure css? If you end up playing around with this I'd love to sync and hear your thoughts 🧠

Reply:
I'm mainly thinking just small perf things. If we can get the browser to do things for us, instead of executing JavaScript that is handy. 

If we're only really concerned with creating a space between distinct children, and we're not fussed on order, maybe we can get some use out of [Column gap](https://developer.mozilla.org/en-US/docs/Web/CSS/column-gap)?